### PR TITLE
pytests for vfu_dma_{map, unmap}_sg

### PR DESCRIPTION
**Pull request by swapnili**
Signed-off-by: Swapnil Ingle <swapnil.ingle@nutanix.com>

### CODE REVIEW

File: test/py/test_map_unmap_sg.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    assert ctx != None
+
+    # Invalid region
+    sg = dma_sg_t()
+    iovec = iovec_t()
+    ret = vfu_map_sg(ctx, sg, iovec)
+    assert ret == -1
+    assert ctypes.get_errno() == errno.EINVAL
+
+    # w/o fd
```
Comment (jlevon) [inline test/py/test_map_unmap_sg.py:48]: can you put this in a separate "test_map_sg_without_fd()" so it shows up separately in the test results please?

File: test/py/test_map_unmap_sg.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        offset=0, addr=0x1000, size=4096)
+    msg(ctx, sock, VFIO_USER_DMA_MAP, payload)
+    sg.region = 0
+    ret = vfu_map_sg(ctx, sg, iovec)
+    assert ret == -1
+    assert ctypes.get_errno() == errno.EFAULT
+
+    disconnect_client(ctx, sock)
+
+	# Multiple sg entries
```
Comment (jlevon) [inline test/py/test_map_unmap_sg.py:63]: same here

**Review by jlevon — COMMENTED** [review]
Thanks for doing this, just had one small nit!

**Comment by jlevon** [comment]
Although, actually, can't you remove the C unit tests? I'd love to get unit_tests.c down to 0 lines

**Comment by swapnili** [comment]
@jlevon addressed your comments please have another look

**Review by jlevon — APPROVED** [review]
thanks!

**Pull request merged**